tests/remote_test.sh

```diff
@@ -127,6 +127,47 @@ function test_populate_remote_info()
   }
 }
 
+function test_is_ssh_connection_configured()
+{
+  local remote='test_remote'
+  local user='test_user'
+  local port='22'
+  local flag='TEST_MODE'
+
+  cd "$TEST_PATH" || {
+    fail "($LINENO) It was not possible to move to temporary directory"
+    return
+  }
+
+  # Case 1: IP, user and port passed as command line arguments (kw ssh -r)
+  remote_parameters['REMOTE_IP']="$remote"
+  remote_parameters['REMOTE_USER']="$user"
+  remote_parameters['REMOTE_PORT']="$port"
+
+  is_ssh_connection_configured "$flag" > /dev/null
+
+  assertEquals "($LINENO):" '0' "$?"
+
+  # Case 2: Using a remote config file
+  remote_parameters['REMOTE_IP']=''
+  remote_parameters['REMOTE_USER']=''
+  remote_parameters['REMOTE_PORT']=''
+  remote_parameters['REMOTE_FILE']="${TEST_PATH}/.kw/remote.config"
+  remote_parameters['REMOTE_FILE_HOST']='origin'
+
+  is_ssh_connection_configured "$flag" > /dev/null
+
+  assertEquals "($LINENO):" '0' "$?"
```

Comment:
Could you create one specific function for checking this case?

Reply:
Do you mean creating another test function for this case? As this case pertains to the `is_ssh_configured()` function, isn't it ok to leave it here? Sorry, I am a little confused 😕

Reply:
Yes, every case here can be a single test. I know the other tests are not like that, but this is something that we will need to change soon on this test file. Ideally, we want one function test per test case to make it easier to debug issues in case of failures. You can have a function test named:

```
function test_is_ssh_connection_configured_with_remote_config_file()
{
...
}

function test_is_ssh_connection_no_remote_config_file()
...
```

Reply:
Oh, I see. Thanks for clarifying this for me. Going to work on that.